packages/base-controller/src/BaseControllerV2.test.ts

```diff
@@ -188,6 +216,23 @@ describe('BaseController', () => {
     );
   });
 
+  it('should allow for applying immer patches to state', () => {
```

Comment:
Actually, one thing I noticed. Should we test that the `stateChange` event gets published when `applyChanges` is called?

Reply:
Done.